lib/src/packet_buffer.dart

```diff
@@ -68,24 +88,32 @@ class PacketBuffer {
         bytes = packet;
       } else {
         // the bytes stradle a boundary
-        flatten();
+        _flatten();
         bytes = _buffer.single;
         packetOffset = _cursor;
       }
       _cursor += length;
-      return bytes.buffer.asUint8List(bytes.offsetInBytes + packetOffset, length);
+      return bytes.buffer
```

Comment:
avoid reformatting in the same patch that makes logic changes, it makes it hard to review the pr